client/src/document/toolbar/flaws.tsx

```diff
@@ -4,14 +4,101 @@ import { annotate, annotationGroup } from "rough-notation";
 import { RoughAnnotation } from "rough-notation/lib/model";
 
 import { humanizeFlawName } from "../../flaw-utils";
-import { Doc, Link, MacroErrorMessage, BadBCDLink } from "../types";
+import {
+  Doc,
+  Link,
+  MacroErrorMessage,
+  BadBCDLink,
+  ImageReferenceFlaw,
+  GenericFlaw,
+} from "../types";
 import "./flaws.scss";
 
 interface FlawCount {
   name: string;
   count: number;
 }
 
+function useAnnotations(genericFlaws: GenericFlaw[]) {
```

Comment:
After *this* lands, I'm going to make the `BrokenLinks` flaw thing use this "hook" too.